frontend/src/components/auth/TeacherSignup/types.tsx

```diff
@@ -0,0 +1,27 @@
+import {
+  FieldErrorsImpl,
+  UseFormHandleSubmit,
+  UseFormRegister,
+} from "react-hook-form";
+
+export interface TeacherSignupForm {
+  firstName: string;
+  lastName: string;
+  email: string;
+  grades: number[];
+  currentlyTeachingJM: boolean;
+  school: string;
+  newSchoolName: string;
+  newSchoolCountry: string;
+  newSchoolCity: string;
+  newSchoolDistrict: string;
+  newSchoolAddress: string;
```

Comment:
thoughts on creating a separate `schoolType` and then setting `school: string | schoolType`?